libs/wingsdk/src/shared-aws/queue.inflight.ts

```diff
@@ -51,10 +51,22 @@ export class QueueClient implements IQueueClient {
   public async approxSize(): Promise<number> {
     const command = new GetQueueAttributesCommand({
       QueueUrl: this.queueUrl,
-      AttributeNames: ["ApproximateNumberOfMessages"],
+      AttributeNames: [
+        "ApproximateNumberOfMessages",
+        "ApproximateNumberOfMessagesNotVisible",
```

Comment:
Including the inflight messages in the count sounds ok to me. Let's note it in the API docstring for `approxSize` (in cloud/queue.ts):

```
Retrieve the approximate number of messages in the queue.
This include messages that are still being processed by other queue clients.
```

Can we add a test for this? For example, if I have a queue consumer that takes 2 minutes to process a message, and I check on the queue's size after 1 minute, the queue's size should still be 1. This might require a change in the simulator.

Reply:
Sure, let me write a specific approxSize test

Reply:
@Chriscbr 

I think I find a problem on queue simulator look at this

https://github.com/winglang/wing/blob/104b550ba1aeb050171a5f561b2a92d80b23c474/libs/wingsdk/src/target-sim/queue.inflight.ts#L20

I don't think that the subscribers to a queue should be an array.
If a queue is consumed by an array of subscribers, it makes its behavior similar to a topic, don't you think?

Reply:
This test is complicated to perform in the simulator because the `processMessages` method of the Queue class in the simulator removes messages from the message list for processing.

Even when avoiding the removal of messages, the problem persists since the execution of the function is done asynchronously. This means that the sleep placed within the function does not prevent the `processMessages` processing from continuing and successfully completes the processing, removing the messages.

This behavior also prevents a function from being interrupted by a `timeout` in the simulator, and essentially, all error handling for the invocation of the `processMessages` function is never executed

I believe that this issue regarding the asynchronous behavior in the simulator is mapped in this issue https://github.com/winglang/wing/issues/1980

Reply:
I see... #1980 seems important for this